test/libsolidity/semanticTests/memoryManagement/return_variable.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    function memorySize() internal pure returns (uint s) {
+        assembly { s := mload(0x40) }
+    }
+    function f() public returns (uint, uint) {
+        uint before = memorySize();
+        g();
+        uint mid = memorySize();
+        h();
+        uint end = memorySize();
+        return (mid - before, end - mid);
+    }
+    function g() internal returns (uint[40] memory) {
```

Comment:
We have to allocate even unnamed return variables in general because of this case. We could avoid allocating if there is a `return x` in all code paths, but I would rather not rely on that check in the code generator. Rather, let us try to do this in the memory allocation optimizer for yul.

Reply:
ok. I think it would be a good idea to put this in a comment because it explains why this test looks like it does.